third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -946,7 +946,7 @@ class ConvertAddPtrOp : public PointerCanonicalizationPattern<tt::AddPtrOp> {
         break;
 
       bool asScalar =
-          isScalarIntConst(uniformSum) && !isScalarIntZero(uniformSum);
+        uniformSum && isScalarIntConst(uniformSum) && !isScalarIntZero(uniformSum);
```

Comment:
Please add a comment at line 910, saying "Note: uniforms could be empty, and hence subsequent uniformSum could be none".